clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java

```diff
@@ -279,204 +260,100 @@ public FilterResult(int messagesRead,
         }
     }
 
-    public static MemoryRecordsBuilder builder(ByteBuffer buffer,
-                                               CompressionType compressionType,
-                                               TimestampType timestampType,
-                                               int writeLimit) {
-        return new MemoryRecordsBuilder(buffer, Record.CURRENT_MAGIC_VALUE, compressionType, timestampType, 0L,
-                System.currentTimeMillis(), 0L, (short) 0, 0, writeLimit);
+    public static MemoryRecords readableRecords(ByteBuffer buffer) {
+        return new MemoryRecords(buffer);
     }
 
-    public static MemoryRecordsBuilder builder(ByteBuffer buffer,
-                                               byte magic,
-                                               CompressionType compressionType,
-                                               TimestampType timestampType,
-                                               int writeLimit) {
-        return new MemoryRecordsBuilder(buffer, magic, compressionType, timestampType, 0L,
-                System.currentTimeMillis(), 0L, (short) 0, 0, writeLimit);
+    public static MemoryRecordsBuilder builder(CompressionType compressionType, long baseOffset) {
+        return builder(compressionType, TimestampType.CREATE_TIME, baseOffset);
     }
 
-    public static MemoryRecordsBuilder builder(ByteBuffer buffer,
-                                               byte magic,
-                                               CompressionType compressionType,
+    public static MemoryRecordsBuilder builder(CompressionType compressionType,
```

Comment:
nit: some of these builder functions are only used for unit test and some others are not called by anyone. Could we move the former to specific unit test classes and remove the latter?